src/duke/task/Task.java

```diff
@@ -18,6 +18,16 @@ public Task(String name, boolean isDone, int index) {
         this.index = index;
     }
 
+    public boolean searchKeyword(String keyword) {
+        String[] s = name.split(" ");
+        for (String ss : s) {
+            if (ss.equals(keyword)) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
Consider adding JavaDoc to all these methods for readability purposes. Speaking of which, it might be necessary to make them inherit or implement a certain class/interface so that you would not have to implement the searchKeyword too many time.